src/open_clip/factory.py

```diff
@@ -56,6 +56,8 @@ def load_state_dict(checkpoint_path: str, map_location='cpu'):
         state_dict = checkpoint
     if next(iter(state_dict.items()))[0].startswith('module'):
         state_dict = {k[7:]: v for k, v in state_dict.items()}
+    if 'positional_embedding' in state_dict:
```

Comment:
Why is this needed

Reply:
Ok i see, checkpoint compatibility
Let's add a comment

Reply:
text tower change will make state dicts from before change and after change incompatible so convert_state_dict fixes that and I think
```
'positional_embeddings' in state_dict
```
just detects that. Please correct me if I'm wrong @rwightman